x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java

```diff
@@ -111,7 +115,16 @@ public EnrichLookupService(
         this.executor = transportService.getThreadPool().executor(EsqlPlugin.ESQL_THREAD_POOL_NAME);
         this.bigArrays = bigArrays;
         this.blockFactory = blockFactory;
-        transportService.registerRequestHandler(LOOKUP_ACTION_NAME, this.executor, LookupRequest::new, new TransportHandler());
+        // The enrich lookup occurs in two steps. First, we send an authentication request, which includes a lookup request,
+        // to 'cluster:monitor/xpack/enrich/esql/lookup' to verify that the current user has the `monitor_enrich` privilege.
+        // Once authenticated, we send the lookup request using 'enrich_origin' to 'indices:data/read/esql/lookup' on the local node.
+        transportService.registerRequestHandler(
+            AUTHENTICATE_ACTION_NAME,
```

Comment:
It would be preferable to use a `org.elasticsearch.xpack.core.security.action.user.HasPrivilegesRequest` to check for privileges. Functionally, it is very similar to what you have here but is a bit more extensible in that you could check for multiple permissions (if needed in the future) and requires less custom code.